CHANGELOG.md

```diff
@@ -8,6 +8,14 @@ and this project adheres to
 
 
 ## [Unreleased]
+The hal.core functionality that implements Representation now requires a `discovery-url-fn` function to be included in the context if you want it to automatically add a discovery url to the resource. This can be achieved by adding the following mixin to your handler:
+
+```clojure
+(defn with-discovery-url-fn [_]
+  {:initialize-context
+   (fn [{:keys [request routes]}]
+     {:discovery-url-fn #(hype/absolute-url-for request routes :discovery)})})
```

Comment:
I have a feeling that you can just bind the :discovery-url-fn directly into the resource rather than have it in an :initialize-context block.

Reply:
The reason I did it like this is we need access to the request which means we need to do it dynamically when the request is handled rather than a start up time when defining the resource.

Alternatively we make passing the request part of the function's API but I thought making the function completely agnostic to the underlying mechanism would be better 🤔